src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextResponse.cs

```diff
@@ -0,0 +1,118 @@
+    public string? ResponseId { get; set; }
+
+    /// <summary>Gets or sets the model ID used in the creation of the speech to text completion.</summary>
+    public string? ModelId { get; set; }
+
+    /// <summary>Gets or sets the raw representation of the speech to text completion from an underlying implementation.</summary>
+    /// <remarks>
+    /// If a <see cref="SpeechToTextResponse"/> is created to represent some underlying object from another object
+    /// model, this property can be used to store that original object. This can be useful for debugging or
+    /// for enabling a consumer to access the underlying object model if needed.
+    /// </remarks>
+    [JsonIgnore]
+    public object? RawRepresentation { get; set; }
+
+    /// <summary>Gets or sets any additional properties associated with the speech to text completion.</summary>
+    public AdditionalPropertiesDictionary? AdditionalProperties { get; set; }
+
+    /// <summary>
+    /// Gets or sets the text of the first <see cref="TextContent"/> instance in <see cref="Contents" />.
+    /// </summary>
+    /// <remarks>
+    /// If there is no <see cref="TextContent"/> instance in <see cref="Contents" />, then the getter returns <see langword="null" />,
+    /// and the setter adds a new <see cref="TextContent"/> instance with the provided value.
+    /// </remarks>
+    [JsonIgnore]
+    public string? Text
+    {
+        get => Contents.OfType<TextContent>().FirstOrDefault()?.Text;
+        set
+        {
+            if (Contents.OfType<TextContent>().FirstOrDefault() is { } textContent)
+            {
+                textContent.Text = value;
+            }
+            else if (value is not null)
+            {
+                Contents.Add(new TextContent(value));
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public override string ToString() => Contents.ConcatText();
+
+    /// <summary>Creates an array of <see cref="SpeechToTextResponseUpdate" /> instances that represent this <see cref="SpeechToTextResponse" />.</summary>
+    /// <returns>An array of <see cref="SpeechToTextResponseUpdate" /> instances that may be used to represent this <see cref="SpeechToTextResponse" />.</returns>
+    public SpeechToTextResponseUpdate[] ToSpeechToTextResponseUpdates()
+    {
+        SpeechToTextResponseUpdate update = new SpeechToTextResponseUpdate
+        {
+            Contents = Contents,
+            AdditionalProperties = AdditionalProperties,
+            RawRepresentation = RawRepresentation,
+            StartTime = StartTime,
+            EndTime = EndTime,
+            Kind = SpeechToTextResponseUpdateKind.TextUpdated,
+            ResponseId = ResponseId,
+            ModelId = ModelId,
+        };
+
+        return [update];
+    }
+
+    /// <summary>Gets or sets the generated content items.</summary>
+    [AllowNull]
+    public IList<AIContent> Contents
+    {
+        get => _contents ??= [];
+        set => _contents = value;
+    }
+}
```

Comment:
Is there an equivalent to Usage for speech to text?

Reply:
No API provided the usage information as part of the request body.